esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6153,32 +6022,86 @@ SecAccessFlags translateToSecAccessFlags(CSecAccessType from)
     }
 }
 
+void CWsDfuEx::dFULookupFileCommon(IEspContext &context, const CDfsLogicalFileName &lfn, const char *requestId, unsigned expirySecs, bool returnTextResponse, IEspDFUFileAccessInfo &accessInfo)
+{
+    StringBuffer fileName;
+    lfn.get(fileName, false, true);
+    if (0 == fileName.length())
+         throw MakeStringException(ECLWATCH_INVALID_INPUT, "No Name defined.");
+
+    StringBuffer userID;
+    context.getUserID(userID);
+
+    Owned<IUserDescriptor> userDesc;
+    if (!userID.isEmpty())
+    {
+        userDesc.setown(createUserDescriptor());
+        userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
+    }
+
+    checkLogicalName(fileName, userDesc, true, false, false, nullptr); // check for read permissions
+
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(fileName, userDesc, false, false, true); // lock super-owners
+    if (!df)
+        throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"Cannot find file '%s'.", fileName.str());
+
+    StringBuffer cluster;
+    lfn.getCluster(cluster);
+    Owned<IFileDescriptor> fileDesc = df->getFileDescriptor(cluster);
+
+    // NB: if file has copies on >1 cluster, they must share the same key
+
+    std::vector<std::string> groups;
+    unsigned numClusters = df->numClusters();
+    for (unsigned c=0; c<numClusters; c++)
+    {
+        StringBuffer clusterName;
+        const char *clusterGroup = df->getClusterName(c, clusterName.clear());
+        groups.push_back(clusterGroup);
+    }
+
+    StringBuffer keyPairName;
+    unsigned port;
+    bool secure;
+    getFileDafilesrvConfiguration(keyPairName, port, secure, fileName, groups);
```

Comment:
It's not an error not to be found here, it would simply mean that this cluster/group has not been configured for security.
It could be that the firewalls/configuration elsewhere means that's okay.
It could also be the case that the meta info was being used, but the physical streams were never used, i.e. the keys were never needed.

It's not until you come to use the meta info to try to connect to a [secure] dafilesrv where you have no keys that it's an issue.

I suspect you're right that could do with clarification somewhere, but I don't think it can go here.
Please open another JIRA re. this, with an example of the error seen when the configuration is wrong and access is denied.